lib/surface/compiler.ex

```diff
@@ -220,18 +221,20 @@ defmodule Surface.Compiler do
   defp convert_node_to_ast(:slot, {_, attributes, children, node_meta}, compile_meta) do
     meta = Helpers.to_meta(node_meta, compile_meta)
 
-    with name when not is_nil(name) and is_atom(name) <-
-           attribute_value(attributes, "name", :default),
-         {:ok, props, _attributes} <-
-           collect_directives([Surface.Directive.SlotProps], attributes, meta) do
+    # TODO: Validate attributes with custom messages
+    name = attribute_value(attributes, "name", :default)
+    index = attribute_value_as_ast(attributes, "index", %Surface.AST.Text{value: 0}, compile_meta)
```

Comment:
Another nice thing to implement `<slot>` as a component is that we could have all this validation for free by just defining `property index, :integer`. Maybe the new `transform` callback along with a few tweaks might allow us to do that soon.

Reply:
Yeah! that would also simplify the compiler and EExEngine code a bit cause we'd be able to remove those cases